src/middlewared/middlewared/scripts/ctdb_events.py

```diff
@@ -0,0 +1,276 @@
+            if srv['enable']:
+                self.client.call('service.restart', srv['name'])
+
+    def shutdown(self):
+        """
+        This shuts down all services managed by ctdb and is
+        triggered by ctdb shutting down.
+        """
+        return
+
+    def monitor(self):
+        if not os.path.exists(CTDB_SERVICE_FILE):
+            return
+
+        if not self.client.call('system.ready'):
+            return
+
+        payload = {
+            'event': 'MONITOR',
+            'status': 'SUCCESS',
+        }
+
+        last_failed = os.path.exists(CTDB_MONITOR_FAILED_SENTINEL)
+
+        try:
+            self.check_ctdb_shared_volume()
+        except Exception as e:
+            payload = {
+                'event': 'MONITOR',
+                'status': 'FAILURE',
+                'reason': str(e),
+                'service': 'ctdb_shared_volume',
+            }
+
+        if not payload['status'] == 'FAILURE':
+            self.load_service_file()
+            for srv in self.cl_services.values():
+                srvinfo = self.client.call('service.query', [['service', '=', srv['name']]], {'get': True})
+                final_state = 'STOPPED'
+                if srv['enable']:
+                    if srvinfo['state'] != 'RUNNING':
+                        self.logger.warning('%s: managed service is not running. Attempting to start.',
+                                            srv['name'])
+                        try:
+                            self.client.call('service.start', srv['name'], {'silent': False})
+                            final_state = 'RUNNING'
+                        except Exception as e:
+                            self.logger.warning('%s: failed to start managed service.',
+                                                srv['name'], exc_info=True)
+
+                            errmsg = f'{srv["name"]}: service failed to start: {e}'
+                            payload = {
+                                'event': 'MONITOR',
+                                'status': 'FAILURE',
+                                'reason': errmsg,
+                                'service': srv['name'],
+                            }
+                            break
+
+                    if not srvinfo['enable']:
+                        self.client.call('service.update', srvinfo['id'], {'enable': True})
+
+                else:
+                    if srvinfo['enable']:
+                        self.client.call('service.update', srvinfo['id'], {'enable': False})
+
+                    if srvinfo['state'] == 'RUNNING':
+                        self.client.call('service.stop', srv['name'])
+
+                self.node_status[srv['name']] = final_state
+
+        if last_failed and payload['status'] == 'SUCCESS':
+            os.remove(CTDB_MONITOR_FAILED_SENTINEL)
+            self.client.call('ctdb.event.process', payload)
+
+        elif payload['status'] == 'FAILURE' and not last_failed:
```

Comment:
Same here as above. If it's failure, just always create the sentinel overwriting it if it already exists?